### PR TITLE
add script to remove core from quant-ph records

### PR DESCRIPTION
**Pull request by MJedr**

### CODE REVIEW

File: scripts/remove-quant-ph/script.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from inspirehep.curation.search_check_do import SearchCheckDo
+
+
+class RemoveQuantPh(SearchCheckDo):
+    """Remove core from quant-ph literature records"""
+
+    query = "arxiv_eprints.categories:quant-ph core:true not _desy_bookkeeping.status:final not _desy_bookkeeping.status:printed"
```
Comment (drjova) [inline scripts/remove-quant-ph/script.py:7]: It has also `earlier_date` in the issue's query, not sure if we need it but without that it affects more records. Could we please double check?

Comment (MJedr) [reply scripts/remove-quant-ph/script.py:7]: I copied the query from the issue 
https://inspirehep.net/literature?sort=mostrecent&size=250&page=1&q=arxiv_eprints.categories%3Aquant-ph%20core%3Atrue%20not%20_desy_bookkeeping.status%3Afinal%20%20not%20_desy_bookkeeping.status%3Aprinted&earliest_date=1930--2022

Comment (MJedr) [reply scripts/remove-quant-ph/script.py]: I don't understand it cause the query in the `list of affected records` doesn't contain a specified date ( 2021-10-21 - 2022-11-16) range. Should I add it? Or is the query correct @michamos ?

File: scripts/remove-quant-ph/script.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class RemoveQuantPh(SearchCheckDo):
+    """Remove core from quant-ph literature records"""
+
+    query = "arxiv_eprints.categories:quant-ph core:true not _desy_bookkeeping.status:final  not _desy_bookkeeping.status:printed and earliest_date:1930--2022"
+
+    @staticmethod
+    def check(record, logger, state):
+        record["core"] is True
```
Comment (michamos) [inline scripts/remove-quant-ph/script.py:11]: it's better to do the full check as in the query here for consistency

File: scripts/remove-quant-ph/script.py (hunk 1 above)
Comment (michamos) [reply scripts/remove-quant-ph/script.py]: the query is almost correct, it's missing the earliest date part which was done as a facet in the link. The filtering by creation date based was an earlier idea that was later refined, so is not needed any longer. You can add `and de < 2023` to reproduce the effects of the earliest date facet filter.

File: scripts/remove-quant-ph/script.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+class RemoveQuantPh(SearchCheckDo):
+    """Remove core from quant-ph literature records"""
+
+    query = "arxiv_eprints.categories:quant-ph core:true not _desy_bookkeeping.status:final  not _desy_bookkeeping.status:printed and de < 2023"
+
+    @staticmethod
+    def check(record, logger, state):
+        arxiv_category_quant_ph = (
+            get_value(record, "arxiv_eprints.categories") == "quant-ph"
```
Comment (michamos) [inline scripts/remove-quant-ph/script.py:17]: the left-hand side is a list of lists (as both `arxiv_eprints` and `categories` are arrays).

File: scripts/remove-quant-ph/script.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        )
+        is_core = record.get("core")
+        desy_bookkeeping_not_final = not record.get("_desy_bookkeeping") == "final"
```
Comment (michamos) [inline scripts/remove-quant-ph/script.py:20]: `_desy_bookkeeping` is an array of objects

File: scripts/remove-quant-ph/script.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        desy_bookkeeping_status_not_printed = (
+            not get_value(record, "_desy_bookkeeping.status") == "printed"
```
Comment (michamos) [inline scripts/remove-quant-ph/script.py:22]: the result of `get_value` here is a list, as `_desy_bookkeeping` is an array.

**Pull request merged**